src/main/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImport.java

```diff
@@ -0,0 +1,266 @@
+        messages[index++] = "Not a valid GenomicsDB workspace: " + workspace;
+      }
+    }
+    
+    if (sizePerColumnPartitionPerSample < 1024L) {
+      messages[index++] = "Size per column partition per sample is tool small." +
+        " Consider using larger size than 10KBytes";
+    }
+
+    return index==0 ? null : messages;
+  }
+
+  /**
+   * Before traversal, fix configuration parameters and initialize
+   * GenomicsDB. Hard-coded to handle only VCF files and headers
+   */
+  @Override
+  public void onTraversalStart() {
+
+    File workspaceDir = new File(workspace);
+
+    if (createWorkspace) {
+
+      int ret = GenomicsDBImporter.createTileDBWorkspace(workspaceDir.getAbsolutePath());
+
+      if (ret > 0) {
+        try {
+          throw new IOException("Directory " + workspaceDir + " already exists");
+        } catch (IOException e) {
+          e.printStackTrace();
+          return;
+        }
+      } else if (ret < 0) {
+        try {
+          throw new IOException("Error creating GenomicsDB workspace");
+        } catch (IOException e) {
+          e.printStackTrace();
+          return;
+        }
+      }
+    }
+
+    initializeIntervals();
+
+    List<VCFHeader> headers = new ArrayList<>();
+
+    Map<String, FeatureReader<VariantContext>> sampleToVCFMap = new HashMap<>();
+    for (String variantFile : variantFileNames) {
+      String variantFileAbsPath = new File(variantFile).getAbsolutePath();
+      AbstractFeatureReader<VariantContext, LineIterator> reader =
+        AbstractFeatureReader.getFeatureReader(variantFileAbsPath, new VCFCodec(), true);
+      headers.add((VCFHeader)reader.getHeader());
+
+      // We assume only one sample per file
+      String sampleName = ((VCFHeader) reader.getHeader()).getGenotypeSamples().get(0);
+      sampleToVCFMap.put(sampleName, reader);
+    }
+
+    Set<VCFHeaderLine> mergedHeader = VCFUtils.smartMergeHeaders(headers, true);
+    sizePerColumnPartitionPerSample *= sampleToVCFMap.size();
+
+    logger.info("Writing data to array - " + workspace + "/" + arrayName);
+
+    try {
+      importer = new GenomicsDBImporter(sampleToVCFMap, mergedHeader,
+        intervals.get(0), workspace, arrayName,
+        sizePerColumnPartitionPerSample, segmentSize,
+        vidMapJSONFile, callsetMapJSONFile);
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * A complete traversal from start to finish. Tool authors who wish to "roll their own" traversal
+   * from scratch can extend this class directly and implement this method. Walker authors should
+   * instead extend a Walker class and implement the Walker-appropriate apply() method, since the
+   * Walker base classes implement the various kinds of traversals for you.
+   */
+  @Override
+  public void traverse() {
+    try {
+      importer.importBatch();
+      logger.info("Current batch written to disk");
+    } catch (IOException e) {
+      e.printStackTrace();
```

Comment:
Throw `UserException` or `GATKException` as appropriate, passing `e` into the constructor, and don't print the stack trace here.